### PR TITLE
Add ExtKeyClientAuth to the ExtKeyUsage field in GenerateSelfSignedCert.

### PR DESCRIPTION
**Pull request by dlorenc**
## Pull Request Guidelines
1. Please read our [contributor guidelines](https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md).
2. See our [developer guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md).
3. Follow the instructions for [labeling and writing a release note for this PR](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes) in the block below.

``` release-note
* Use the release-note-* labels to set the release note state 
* Clear this block to use the PR title as the release note 
-OR-
* Enter your extended release note here
```

[Analytics]()

This change allows the generated certificates to be used for both client and server authentication. They can then be used by kubecfg when talking to the API server.

### CODE REVIEW

**Comment by dlorenc** [comment]
cc @luxas

File: pkg/util/crypto/crypto.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func GenerateSelfSignedCert(host, certPath, keyPath string, alternateIPs []net.I
 		NotAfter:  time.Now().Add(time.Hour * 24 * 365),
 
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
```
Comment (liggitt) [inline pkg/util/crypto/crypto.go]: I would not expect a serving cert to be used as a client cert. If we want to generate those, we should do so separately.

Comment (dlorenc) [inline pkg/util/crypto/crypto.go]: I think for that to work we'd need to split this cert up into a CA cert and the server cert. Then we'd need to build the client and server certs derived from that one. 

According to my reading of the RFC though, this type of usage is completely fine and this field is optional: http://tools.ietf.org/html/rfc5280#section-4.2.1.12

Basically, leaving it off completely allows the cert to be used for anything, but including the Server flag limits it to just ServerAuth.

Comment (liggitt) [inline pkg/util/crypto/crypto.go]: There purpose of this function is to make a self signed serving cert. if we want certs for clients, thats fine, but I wouldn't combine it here.

Comment (dlorenc) [inline pkg/util/crypto/crypto.go]: Sounds good, I could parameterize this function or make a second one for clients. Which would you prefer?

Comment (liggitt) [inline pkg/util/crypto/crypto.go]: Second one. "Host" makes less sense for a client cert, it's typically just a subject.

**Pull request closed without merging**